runelite-client/src/main/java/net/runelite/client/plugins/lmsperformancetracker/Fighter.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2019, Matsyir <https://github.com/Matsyir>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.lmsperformancetracker;
+
+import lombok.Getter;
+import net.runelite.api.Player;
+
+@Getter
+public class Fighter
+{
+	private Player player;
+	private String name; // username
+	private int attackCount; // total number of attacks
+	private int successCount; // total number of successful attacks
+	private double successRate; // success rate in percentage (0-100)
+	private boolean dead; // will be true if the fighter died in the fight
+	private boolean currentlyAttacking; // will be true if the player is currently doing an attack animation
+
+	// fighter that is bound to a player and gets updated during a fight
+	public Fighter(Player player)
+	{
+		this.player = player;
+		name = player.getName();
+		attackCount = 0;
+		successCount = 0;
+		successRate = 0;
+		dead = false;
+		currentlyAttacking = false;
+	}
+
+	// create a basic Fighter to only hold stats, for the TotalStatsPanel,
+	// but not actually updated during a fight.
+	public Fighter(String name)
```

Comment:
Well, it's updated, but not _during a fight_. New stats get added to it after a fight is completed. If I updated it during a fight, I'd have to constantly update the fighter that holds total stats alongside the actual fight's player. Thinking of how that would be done, it seems pretty messy. I'm also not sure what the behaviour would be if you switched accounts. It might break or the name might become outdated. I'll look into doing this but I think a lot would have to be changed. I don't think it would be that useful, you probably won't be looking at the total stats panel while you're focused on a fight.